src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -88,14 +88,54 @@ namespace System.Text
 #endif
     public abstract class Encoding : ICloneable
     {
-        private static volatile Encoding defaultEncoding;
-        private static volatile Encoding unicodeEncoding;
-        private static volatile Encoding bigEndianUnicode;
-        private static volatile Encoding utf7Encoding;
-        private static volatile Encoding utf8Encoding;
-        private static volatile Encoding utf32Encoding;
-        private static volatile Encoding asciiEncoding;
-        private static volatile Encoding latin1Encoding;
+        // We lazily-initialize the static Encoding properties by storing them in
+        // nested static classes. The static constructor, which holds the initialization
+        // logic, will only be run on the first access to a static member of the class
+        // (see dotnet/coreclr#1193 for more info). Thus, we get the benefits of both
+        // thread-safe and lock-free initialization.
+
+        // Please do not aggregate these classes into one static class. Otherwise,
+        // all of these encodings will be initialized when only one of them is needed.
+
+        private static class ASCIIHolder
+        {
```

Comment:
I do not see a problem with internal static field that is not used anywhere else in the file. It is pretty common with the existing uses of this pattern.

One example out of many:
https://github.com/dotnet/coreclr/blob/775003a4c72f0acc37eab84628fcef541533ba4e/src/mscorlib/src/System/Diagnostics/Eventing/TraceLogging/TraceLoggingDataCollector.cs#L27